web/src/OverviewTable.test.tsx

```diff
@@ -35,3 +49,170 @@ it("shows buttons on the appropriate resources", () => {
 
   expect(actualButtons).toEqual(expectedButtons)
 })
+
+describe("overview table with groups", () => {
+  let view: Proto.webviewView
+  let wrapper: ReactWrapper<OverviewTableProps, typeof TableGroupedByLabels>
+
+  beforeEach(() => {
+    view = nResourceWithLabelsView(5)
+    wrapper = mount(<TableGroupedByLabels view={view} />)
+
+    mockAnalyticsCalls()
+  })
+
+  afterEach(() => cleanupMockAnalyticsCalls())
+
+  // TODO: When resource groups are live in table view, add tests for:
+  //       If no resources have labels, it renders a single table view
+  //       If labels are not enabled, it renders a single table view
+  //       If there are labels and feature is enabled, it renders table with groups
+
+  describe("display", () => {
+    let resources: GroupByLabelView<RowValues>
+
+    beforeEach(() => {
+      resources = resourcesToTableCells(
+        view.uiResources,
+        view.uiButtons,
+        new LogStore()
+      )
+    })
+
+    it("renders each label group in order", () => {
+      const { labels: sortedLabels } = resources
+      const groupNames = wrapper.find(OverviewGroupName)
+
+      // Loop through the sorted labels (which includes every label
+      // attached to a resource, but not unlabeled or tiltfile "labels")
+      sortedLabels.forEach((label, idx) => {
+        const groupName = groupNames.at(idx)
+        expect(groupName.text()).toBe(label)
+      })
+    })
+
+    // Note: the sample data generated in the test helper `nResourcesWithLabels`
+    // always includes unlabeled resources
+    it("renders a resource group for unlabeled resources and for Tiltfiles", () => {
+      const groupNames = wrapper.find(OverviewGroupName)
+
+      const unlabeledLabel = groupNames.filterWhere(
+        (name) => name.text() == UNLABELED_LABEL
+      )
+      expect(unlabeledLabel.length).toBe(1)
+
+      const tiltfileLabel = groupNames.filterWhere(
+        (name) => name.text() == TILTFILE_LABEL
+      )
+      expect(tiltfileLabel.length).toBe(1)
+    })
+
+    it("renders a table for each resource group", () => {
+      const tables = wrapper.find(Table)
+      const totalLabelCount =
+        resources.labels.length +
+        (resources.tiltfile.length ? 1 : 0) +
+        (resources.unlabeled.length ? 1 : 0)
+
+      expect(tables.length).toBe(totalLabelCount)
+    })
+
+    it("renders the correct resources in each label group", () => {
+      const { labelsToResources, unlabeled, tiltfile } = resources
+      const resourceGroups = wrapper.find(OverviewGroup)
+
+      resourceGroups.forEach((group) => {
+        // Find the label group name
+        const groupName = group.find(OverviewGroupName).text()
+        // Find the resource list displayed in the table
+        const table = group.find(Table)
+        const resourcesInTable = table
+          .find(TableNameColumn)
+          .map((resourceName) => resourceName.text())
```

Comment:
yeah, that's a good point. i hesitate to hardcode expected cases based on shared mocked data because, (at least in other codebases i've worked on), when that data is changed for an unrelated reason, it can cause a bunch of tests to falsely fail. (though one might argue that false negatives are better than false positives when it comes to tests passing!)

i think in this case, because the `toEqual()` values are generated from the `view`, we should get a correct failure if `OverviewTable` changes. i'll keep this in mind for future tests i write.

Reply:
Oops, you're right, I got my expected and actual reversed. Sorry!

(maybe I've become too reliant on the pattern of explicitly naming expected and actual vars "expected" and "actual")

Reply:
gotchu! it'll be helpful if i rename the vars for better clarity.